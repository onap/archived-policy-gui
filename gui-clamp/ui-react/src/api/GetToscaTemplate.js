/*-
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * ============LICENSE_END=========================================================
 */
import React, { useState } from "react";
import Button from "react-bootstrap/Button";

const GetToscaTemplate = (props) => {

  const [windowLocationPathName, setWindowLocationPathname] = useState('');

  const getTemplateHandler = async () => {
    console.log('getTemplateHandler called')
    setWindowLocationPathname(window.location.pathname);

    const params = {
      name: props.templateName,
      version: props.templateVersion
    }

    const response = await fetch(windowLocationPathName +
      '/restservices/clds/v2/toscaControlLoop/getToscaTemplate' + '?' + (new URLSearchParams(params)));

    const data = await response.json();

    props.onGetToscaServiceTemplate(data);

  }

  return (
    <React.Fragment>
      <Button variant="primary"
              type="submit"
              onClick={ getTemplateHandler }>Get Tosca Service Template</Button>
    </React.Fragment>
  );


}

export default GetToscaTemplate;
