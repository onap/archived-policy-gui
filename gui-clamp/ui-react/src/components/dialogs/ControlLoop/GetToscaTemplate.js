/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import React, { useState } from "react";
import Button from "react-bootstrap/Button";
import ControlLoopService from "../../../api/ControlLoopService";

const GetToscaTemplate = (props) => {

  const getTemplateHandler = async () => {

    const response = await ControlLoopService.getToscaTemplate(props.templateName, props.templateVersion)
      .catch(error => error.message);

    if(!response.ok) {
      console.log('getToscaServiceTemplateHandler called with error')

    } else
    {
      console.log('getToscaServiceTemplateHandler called')
    }

    props.onGetToscaServiceTemplate(response);

  }

  return (
    <React.Fragment>
      <Button variant="primary"
              type="submit"
              onClick={ getTemplateHandler }>Pull Tosca Service Template</Button>
    </React.Fragment>
  );


}

export default GetToscaTemplate;
