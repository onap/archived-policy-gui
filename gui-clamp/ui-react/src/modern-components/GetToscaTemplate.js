import React from "react";
import Button from "react-bootstrap/Button";

const GetToscaTemplate = (props) => {

  const getTemplateHandler = async () => {

    const params = {
      name: props.templateName,
      version: props.templateVersion
    }

    const response = await fetch('https://localhost:3000' +
      '/onap/controlloop/v2/commission/' + 'toscaservicetemplate'
      + '?' + (new URLSearchParams(params)), {
      headers: {
        'Authorization': 'Basic ' + btoa('healthcheck' + ':' + 'zb!XztG34')
      }
    });

    const data = await response.json();

    props.onGetToscaServiceTemplate(data);

  }

  return (
    <React.Fragment>
      <Button variant="primary"
              type="submit"
              onClick={ getTemplateHandler }>Get Tosca Service Template</Button>
    </React.Fragment>
  );


}

export default GetToscaTemplate;