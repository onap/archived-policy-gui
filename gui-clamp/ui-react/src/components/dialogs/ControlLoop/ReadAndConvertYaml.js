/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import React, { useState } from "react";
import GetToscaTemplate from "./GetToscaTemplate";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";
import { Alert } from "react-bootstrap";

import styled from 'styled-components';
import DeleteToscaTemplate from "./DeleteToscaTemplate";

const ModalStyled = styled(Modal)`
  background-color: transparent;
`

const AlertStyled = styled(Alert)`
  margin-top: 10px;
`

const PreStyled = styled.pre`
  color: #7F0055;
  overflow: auto;
  max-height: 70vh;
  margin-top: 10px;
`

const ReadAndConvertYaml = (props) => {
  const [show, setShow] = useState(true);
  const [toscaTemplateData, setToscaTemplateData] = useState();
  const [deleteToscaTemplateData, setDeleteToscaTemplateData] = useState();
  const [responseOk, setResponseOk] = useState(true);
  const [deleteResponseOk, setDeleteResponseOk] = useState(true);
  const name = 'ToscaServiceTemplateSimple';
  const version = '1.0.0';

  const handleClose = () => {
    console.log('handleClose called');
    setShow(false);
    props.history.push('/');
  }

  const getToscaServiceTemplateHandler = async (toscaServiceTemplateResponse) => {
    // console.log('getToscaServiceTemplateHandler called: ' + toscaServiceTemplate);

    if (!toscaServiceTemplateResponse.ok) {
      console.log('Response is not ok');
      setResponseOk(false);
      const toscaData = await toscaServiceTemplateResponse.text();
      setToscaTemplateData(toscaData);
    } else {
      setResponseOk(true);
      const toscaData = await toscaServiceTemplateResponse.json();
      setToscaTemplateData(toscaData);
    }
  }

  const deleteToscaServiceTemplateHandler = async (deleteToscaServiceTemplateResponse) => {

    if (!deleteToscaServiceTemplateResponse.ok) {
      console.log('Delete response not ok');
      setDeleteResponseOk(false);
      const deleteToscaData = await deleteToscaServiceTemplateResponse;
      setDeleteToscaTemplateData(deleteToscaData);
    } else {
      setDeleteResponseOk(true);
      const deleteToscaData = await deleteToscaServiceTemplateResponse.json();
      setDeleteToscaTemplateData(deleteToscaData);
    }
  }

  return (
    <ModalStyled size="xl"
                 show={ show }
                 onHide={ handleClose }
                 backdrop="static"
                 keyboard={ false }>
      <Modal.Header closeButton>
        <Modal.Title>View Tosca Template</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <GetToscaTemplate templateName={ name }
                          templateVersion={ version }
                          onGetToscaServiceTemplate={ getToscaServiceTemplateHandler }/>
        { responseOk && <PreStyled> { JSON.stringify(toscaTemplateData, null, 2) } </PreStyled> }
        <AlertStyled show={ !responseOk }
                     variant="danger">{ toscaTemplateData }</AlertStyled>
        { deleteResponseOk && responseOk && toscaTemplateData != null &&
        <DeleteToscaTemplate templateName={ name }
                             templateVersion={ version }
                             onDeleteToscaServiceTemplate={ deleteToscaServiceTemplateHandler }
        />
        }
        <AlertStyled show={ !deleteResponseOk }
                     variant="danger">{ deleteToscaTemplateData }</AlertStyled>
        <AlertStyled show={ deleteResponseOk && deleteToscaTemplateData != null }
                     variant="success"><h2>Delete Successful</h2><PreStyled>{ JSON.stringify(deleteToscaTemplateData, null, 2) }</PreStyled></AlertStyled>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="secondary"
                type="null"
                onClick={ handleClose }>Close</Button>
      </Modal.Footer>
    </ModalStyled>
  );
}
;

export default ReadAndConvertYaml;
