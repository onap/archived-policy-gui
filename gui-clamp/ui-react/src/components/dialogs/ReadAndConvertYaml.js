/*-
 * ============LICENSE_START=======================================================
 *  Copyright (C) 20XX-20YY Nordix Foundation.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 * ============LICENSE_END=========================================================
 */
import React, { useState } from "react";
import GetToscaTemplate from "../../api/GetToscaTemplate";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";

import styled from 'styled-components';

const ModalStyled = styled(Modal)`
  background-color: transparent;
`

const ErrMsgStyled = styled.div`
  color: red;
`

const PreStyled = styled.pre`
  color: #7F0055;
  overflow: auto;
  max-height: 70vh;
`

const ReadAndConvertYaml = (props) => {
  const [show, setShow] = useState(true);
  const [toscaTemplateData, setToscaTemplateData] = useState();
  const name = 'ToscaServiceTemplateSimple';
  const version = '1.0.0';

  const handleClose = () => {
    setShow(false);
    props.history.push('/');
  }

  const getToscaServiceTemplateHandler = (toscaServiceTemplate) => {
    const toscaData = {
      ...toscaServiceTemplate,
      id: Math.random().toString()
    };
    console.log(toscaData);
    setToscaTemplateData(toscaData);
  }

  return (
    <ModalStyled size="xl"
                 show={ show }
                 onHide={ handleClose }
                 backdrop="static"
                 keyboard={ false }>
      <Modal.Header closeButton>
        <Modal.Title>View Tosca Template</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <GetToscaTemplate templateName={ name }
                          templateVersion={ version }
                          onGetToscaServiceTemplate={ getToscaServiceTemplateHandler }/>
        <PreStyled>{ JSON.stringify(toscaTemplateData, null, 2) }</PreStyled>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="secondary"
                type="null"
                onClick={ handleClose }>Cancel</Button>
      </Modal.Footer>
    </ModalStyled>
  );
}

export default ReadAndConvertYaml;
