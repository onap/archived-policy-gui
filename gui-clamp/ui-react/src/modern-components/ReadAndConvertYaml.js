
import React, { useState } from "react";
import GetToscaTemplate from "./GetToscaTemplate";
import Modal from "react-bootstrap/Modal";
import Button from "react-bootstrap/Button";

import styled from 'styled-components';

const ModalStyled = styled(Modal)`
  background-color: transparent;
`

const ErrMsgStyled = styled.div`
  color: red;
`

const PreStyled = styled.pre`
  color: #7F0055;
  overflow: auto;
  max-height: 70vh;
`

const ReadAndConvertYaml = (props) => {
  const [show, setShow] = useState(true);
  const [toscaTemplateData, setToscaTemplateData] = useState();
  const name = 'ToscaServiceTemplateSimple';
  const version = '1.0.0';

  const handleClose = () => {
    setShow(false);
    props.history.push('/');
  }

  const getToscaServiceTemplateHandler = (toscaServiceTemplate) => {
    const toscaData = {
      ...toscaServiceTemplate,
      id: Math.random().toString()
    };
    console.log(toscaData);
    setToscaTemplateData(toscaData);
  }

  return (
    <ModalStyled size="xl"
                 show={ show }
                 onHide={ handleClose }
                 backdrop="static"
                 keyboard={ false }>
      <Modal.Header closeButton>
        <Modal.Title>View Tosca Template</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <GetToscaTemplate templateName={ name }
                          templateVersion={ version }
                          onGetToscaServiceTemplate={ getToscaServiceTemplateHandler }/>
        <PreStyled>{ JSON.stringify(toscaTemplateData, null, 2) }</PreStyled>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="secondary"
                type="null"
                onClick={ handleClose }>Cancel</Button>
      </Modal.Footer>
    </ModalStyled>
  );
}

export default ReadAndConvertYaml;