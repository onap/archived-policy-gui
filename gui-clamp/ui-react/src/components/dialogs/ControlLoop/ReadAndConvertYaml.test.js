/*
 * ============LICENSE_START=======================================================
 *  Copyright (C) 2021 Nordix Foundation.
 *  ================================================================================
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 *  SPDX-License-Identifier: Apache-2.0
 *  ============LICENSE_END=========================================================
 */

import React from 'react';
import { mount, shallow } from 'enzyme';
import ReadAndConvertYaml from './ReadAndConvertYaml';
import GetToscaTemplate from "./GetToscaTemplate";
import toJson from "enzyme-to-json";
import { act } from "react-dom/test-utils";
import { createMemoryHistory } from "history";

let logSpy = jest.spyOn(console, 'log')
describe('Verify ReadAndConvertYaml', () => {

  const unmockedFetch = global.fetch

  const flushPromises = () => new Promise(setImmediate);

  beforeAll(() => {
    global.fetch = () =>
      Promise.resolve({
        ok: true,
        status: 200,
        text: () => "OK",
        json: () => Promise.resolve({
          "tosca_definitions_version": "tosca_simple_yaml_1_1_0",
          "data_types": {},
          "policy_types": {},
          "topology_template": {},
          "name": "ToscaServiceTemplateSimple",
          "version": "1.0.0",
          "metadata": {},
          "id": "0.19518677404255147"
        })
      })
  })

  beforeEach(() => {
    logSpy.mockClear()
  })

  afterAll(() => {
    global.fetch = unmockedFetch
  })

  it("renders without crashing", () => {
    shallow(<ReadAndConvertYaml/>);
  });

  it("renders correctly", () => {
    const tree = shallow(<ReadAndConvertYaml/>);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should have a GetToscaTemplate element', () => {
    const container = shallow(<ReadAndConvertYaml/>)
    expect(container.find('GetToscaTemplate').length).toEqual(1);
  });

  it('should call getToscaServiceTemplateHandler on click', async () => {
    const component = mount(<ReadAndConvertYaml/>);

    await act(async () => {
      component.find('GetToscaTemplate').simulate('click');
      await flushPromises()
      component.update()
      await expect(logSpy).toHaveBeenCalledWith('getToscaServiceTemplateHandler called');
    });
    component.unmount()
  });

  it('should make unsuccessful call getToscaServiceTemplateHandler on click', async () => {
    jest
      .spyOn(global, 'fetch')
      .mockImplementationOnce(async () =>
          Promise.resolve({
            ok: false,
            status: 200,
            json: () => {
              return Promise.resolve({
                "tosca_definitions_version": "tosca_simple_yaml_1_1_0",
                "data_types": {},
                "policy_types": {},
                "topology_template": {},
                "name": "ToscaServiceTemplateSimple",
                "version": "1.0.0",
                "metadata": {},
                "id": "0.19518677404255147"
              })
            }
          }
        )
      )
    const component = mount(<ReadAndConvertYaml/>);

    await act(async () => {
      component.find('GetToscaTemplate').simulate('click');
      await flushPromises()
      component.update()
      expect(logSpy).toHaveBeenCalledWith('getToscaServiceTemplateHandler called with error');
    });
    component.unmount();
  });

  it('should make unsuccessful call deleteToscaServiceTemplateHandler on click', async () => {
    jest
      .spyOn(global, 'fetch')
      .mockImplementationOnce(() =>
        Promise.resolve({
            ok: true,
            status: 200,
            json: () => {
              return Promise.resolve({
                "tosca_definitions_version": "tosca_simple_yaml_1_1_0",
                "data_types": {},
                "policy_types": {},
                "topology_template": {},
                "name": "ToscaServiceTemplateSimple",
                "version": "1.0.0",
                "metadata": {},
                "id": "0.19518677404255147"
              })
            }
          }
        )
      )
    const component = mount(<ReadAndConvertYaml/>);

    await act(async () => {
      component.find('GetToscaTemplate').simulate('click');
      await flushPromises()
      component.update()
      component.find('DeleteToscaTemplate').simulate('click');
      await flushPromises()
      component.update()
      expect(logSpy).toHaveBeenCalledWith('deleteTemplateHandler called');
    });

    component.unmount()
  });

  it('should make unsuccessful call deleteToscaServiceTemplateHandler on click', async () => {
    const realUseState = React.useState
    const stubInitialState = [true]

    const useStateSpy = jest.spyOn(React, 'useState')
    useStateSpy.mockImplementationOnce(() => realUseState(stubInitialState));

    jest
      .spyOn(global, 'fetch')
      .mockImplementationOnce(() =>
        Promise.resolve({
            ok: true,
            status: 200,
            json: () => {
              return Promise.resolve({
                "tosca_definitions_version": "tosca_simple_yaml_1_1_0",
                "data_types": {},
                "policy_types": {},
                "topology_template": {},
                "name": "ToscaServiceTemplateSimple",
                "version": "1.0.0",
                "metadata": {},
                "id": "0.19518677404255147"
              })
            }
          }
        )
      )
    const component = mount(<ReadAndConvertYaml/>);


    await act(async () => {
      component.find('GetToscaTemplate').simulate('click');
      await flushPromises()
      component.update()
    });

    jest
      .spyOn(global, 'fetch')
      .mockImplementationOnce(() =>
        Promise.resolve({
            ok: false,
            status: 200,
            json: () => {
              return Promise.resolve({
                "tosca_definitions_version": "tosca_simple_yaml_1_1_0",
                "data_types": {},
                "policy_types": {},
                "topology_template": {},
                "name": "ToscaServiceTemplateSimple",
                "version": "1.0.0",
                "metadata": {},
                "id": "0.19518677404255147"
              })
            }
          }
        )
      )

    await act(async () => {
      component.find('DeleteToscaTemplate').simulate('click');
      await flushPromises()
      component.update()
      expect(logSpy).toHaveBeenCalledWith('deleteTemplateHandler called with error');
    });
    component.unmount()
  });

  it('handleClose called when bottom button clicked', () => {
    const history = createMemoryHistory();
    const component = mount(<ReadAndConvertYaml history={ history }/>)


    act(() => {
      component.find('[variant="secondary"]').simulate('click');
      expect(logSpy).toHaveBeenCalledWith('handleClose called');
    });
    component.unmount()
  });

  it('handleClose called when top-right button clicked', async () => {
    const history = createMemoryHistory();
    const component = mount(<ReadAndConvertYaml history={ history }/>)
    const logSpy = jest.spyOn(console, 'log');


    await act(async () => {
      component.find('[size="xl"]').get(0).props.onHide();
      await flushPromises()
      component.update()
      expect(logSpy).toHaveBeenCalledWith('handleClose called');
    });
    component.unmount()
  });
});
