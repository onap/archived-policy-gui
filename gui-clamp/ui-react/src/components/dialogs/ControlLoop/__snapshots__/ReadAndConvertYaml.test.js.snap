// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify ReadAndConvertYaml renders correctly 1`] = `
<Styled(Modal)
  backdrop="static"
  keyboard={false}
  onHide={[Function]}
  show={true}
  size="xl"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      View Tosca Template
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <GetToscaTemplate
      onGetToscaServiceTemplate={[Function]}
      templateName="ToscaServiceTemplateSimple"
      templateVersion="1.0.0"
    />
    <styled.pre>
       
       
    </styled.pre>
    <Styled(Alert)
      show={false}
      variant="danger"
    />
    <Styled(Alert)
      show={false}
      variant="danger"
    />
    <Styled(Alert)
      show={false}
      variant="success"
    >
      <h2>
        Delete Successful
      </h2>
      <styled.pre />
    </Styled(Alert)>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="null"
      variant="secondary"
    >
      Close
    </Button>
  </ModalFooter>
</Styled(Modal)>
`;
